modules/@apostrophecms/widget-type/ui/apos/mixins/AposWidgetMixin.js

```diff
@@ -52,21 +52,32 @@ export default {
   methods: {
     async renderContent() {
       apos.bus.$emit('widget-rendering');
+      const {
+        aposLivePreview,
+        ...widget
+      } = this.modelValue;
       const parameters = {
         _docId: this.docId,
-        widget: this.modelValue,
+        widget,
         areaFieldId: this.areaFieldId,
-        type: this.type
+        type: this.type,
+        livePreview: aposLivePreview
       };
       try {
         if (this.rendering && (isEqual(this.rendering.parameters, parameters))) {
           this.rendered = this.rendering.html;
         } else {
-          this.rendered = '...';
-          this.rendered = await apos.http.post(`${apos.area.action}/render-widget?aposEdit=1&aposMode=${this.mode}`, {
+          // Don't use a placeholder here, it causes flickering in live preview mode.
+          // It is better to display the old until we display the new,
+          // we have "busy" for clarity
+          const result = await apos.http.post(`${apos.area.action}/render-widget?aposEdit=1&aposMode=${this.mode}`, {
             busy: true,
             body: parameters
           });
+          //
+          if (result !== 'aposLivePreviewSchemaNotYetValid') {
```

Comment:
If we are previewing and the schema isn't valid at the moment according to the server, we have nothing to do here, leave the previous placeholder or preview alone.